File: pyg_shop_web/src/main/webapp/js/service/itemCatService.js
//服务层
app.service('itemCatService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../itemCat/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../itemCat/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../itemCat/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../itemCat/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../itemCat/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../itemCat/deleteAll.do?ids='+ids);
	}
	//搜索
/*	this.search=function(page,rows,searchEntity){
		return $http.post('../itemCat/search.do?page='+page+"&rows="+rows, searchEntity);
	}*/
    this.search=function(page,rows,id){
        return $http.post('../itemCat/search.do?page='+page+"&rows="+rows+"&id="+id );
    }

	//根据父ID查找分类信息
    this.findByPid = function (id) {
		return $http.get("../itemCat/findByPid.do?id="+id);
    }
    

});

File: pyg_sellergoods_interface/src/main/java/com/pyg/sellergoods/service/ItemCatService.java
package com.pyg.sellergoods.service;
import java.util.List;
import java.util.Map;

import com.pyg.entity.PageResult;
import com.pyg.pojo.TbItemCat;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface ItemCatService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbItemCat> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TbItemCat itemCat);
	
	
	/**
	 * 修改
	 */
	public void update(TbItemCat itemCat);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbItemCat findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);
	public void deleteAll(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(long id, int pageNum, int pageSize);

	/**
	* @author 沈小云
	* @description  根据父ID查询分类信息
	* @date 2018/10/13 10:13
	* @param
	*@return com.pyg.pojo.TbItemCat
	*/
	public List<TbItemCat> getItemCatByPid(Long id);

	List<Map> selectCatList(Long id);
}
